src/main/scala/com/microsoft/hyperspace/index/IndexLogEntry.scala

```diff
@@ -111,6 +102,43 @@ object Content {
       None
     }
   }
+
+  /**
+   * Apply `func` to each file in directory recursively.
+   * @param prefixPath Root prefix
+   * @param directory Root directory
+   * @param func function which would apply to current prefix and file
+   * @tparam T
+   * @return
+   */
+  def recFilesApply[T](
+              prefixPath: Path,
+              directory: Directory,
+              func: (FileInfo, Path) => T): Seq[T] = {
+    @tailrec
+    def recAcc[A](
+                   dirMap: List[(Path, Seq[Directory])],
+                   func: (FileInfo, Path) => A,
+                   acc: Seq[A] = Seq.empty): Seq[A] = {
+      dirMap match {
+        case Nil => acc
+        case (curPrefixPath, curDirs) :: xs =>
```

Comment:
Could you explain why it's `xs`?

Reply:
I have started with common unapplied list ```x::xs``` and simplified head later)
xs is plural for x.

I will rename it `otherDirs`  for readability.

Reply:
fixed